config/config_test.go

```diff
@@ -0,0 +1,333 @@
+	cases := map[string]struct {
+		cfg       *Config
+		taskMeta  awsutil.ECSTaskMeta
+		expConfig func(awsutil.ECSTaskMeta) discovery.Config
+	}{
+		"basic flags without TLS or ACLs": {
+			cfg: &Config{
+				ConsulServers: ConsulServers{
+					Hosts:    "consul.dc1.address",
+					GRPCPort: 8503,
+				},
+			},
+			expConfig: func(t awsutil.ECSTaskMeta) discovery.Config {
+				return discovery.Config{
+					Addresses: "consul.dc1.address",
+					GRPCPort:  8503,
+				}
+			},
+		},
+		"basic flags without TLS or ACLs with custom gRPC port": {
+			cfg: &Config{
+				ConsulServers: ConsulServers{
+					Hosts:    "consul.dc1.address",
+					GRPCPort: 8502,
+				},
+			},
+			expConfig: func(t awsutil.ECSTaskMeta) discovery.Config {
+				return discovery.Config{
+					Addresses: "consul.dc1.address",
+					GRPCPort:  8502,
+				}
+			},
+		},
+		"default TLS": {
+			cfg: &Config{
+				ConsulServers: ConsulServers{
+					Hosts:     "consul.dc1.address",
+					GRPCPort:  8503,
+					EnableTLS: true,
+				},
+			},
+			expConfig: func(t awsutil.ECSTaskMeta) discovery.Config {
+				return discovery.Config{
+					Addresses: "consul.dc1.address",
+					GRPCPort:  8503,
+					TLS:       &tls.Config{},
+				}
+			},
+		},
+		"ACL Auth method": {
+			cfg: &Config{
+				ConsulServers: ConsulServers{
+					Hosts: "consul.dc1.address",
+				},
+				ConsulLogin: ConsulLogin{
+					Enabled:    true,
+					Method:     "test-auth-method",
+					Datacenter: "test-dc",
+					Meta:       map[string]string{"key1": "value1", "key2": "value2"},
+				},
+			},
+			taskMeta: awsutil.ECSTaskMeta{
+				Cluster: "test-cluster",
+				TaskARN: "arn:aws:ecs:us-east-1:123456789:task/test/abcdef",
+				Family:  "family-service",
+			},
+			expConfig: func(t awsutil.ECSTaskMeta) discovery.Config {
+				return discovery.Config{
+					Addresses: "consul.dc1.address",
+					Credentials: discovery.Credentials{
+						Type: discovery.CredentialsTypeLogin,
+						Login: discovery.LoginCredential{
+							AuthMethod: "test-auth-method",
+							Datacenter: "test-dc",
+							Meta: map[string]string{
+								"key1":                         "value1",
+								"key2":                         "value2",
+								"consul.hashicorp.com/task-id": t.TaskID(),
+								"consul.hashicorp.com/cluster": t.Cluster,
+							},
+						},
+					},
+				}
+			},
+		},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			var srvConfig testutil.ServerConfigCallback
+			if c.cfg.ConsulLogin.Enabled {
+				if testutil.EnterpriseFlag() {
+					c.cfg.Service.Namespace = "test-namespace"
+					c.cfg.Service.Partition = "test-partition"
+				}
+
+				// Enable ACLs to test with the auth method
+				srvConfig = testutil.ConsulACLConfigFn
+				_, apiCfg := testutil.ConsulServer(t, srvConfig)
+				consulClient, err := api.NewClient(apiCfg)
+				require.NoError(t, err)
+				fakeAws := testutil.AuthMethodInit(t, consulClient, c.taskMeta.Family, c.cfg.ConsulLogin.Method)
```

Comment:
In this test, we should always create the auth method in the default namespace, regardless of where the namespace service/task is registering into (this is what the ACL controller does).

Reply:
Removed the custom partition and namespace for enterprise usecases